### PR TITLE
[Serving] Support TIR kernel for PagedKVCache in Llama

### PR DESCRIPTION
**Pull request by MasterJH5574**
This PR supports TIR-kernel-based PagedKVCache modeling and uses it for Llama.

Co-authored-by: Hongyi Jin <jinhongyi02@gmail.com>
Co-authored-by: Bohan Hou <spectrometerh@gmail.com>

### CODE REVIEW

**Comment by junrushao** [comment]
Please rebase and I'd love to review

**Comment by MasterJH5574** [comment]
@junrushao Just rebased, thank you!

File: python/mlc_chat/model/llama/llama_model.py [hunk 1]
```diff
@@ -355,6 +355,24 @@ def create_flashinfer_paged_kv_cache(
             dtype=self.dtype,
         )
 
+    def create_tir_paged_kv_cache(
+        self, max_batch_size: tir.Var, max_total_seq_len: tir.Var, page_size: tir.Var
+    ) -> PagedKVCache:
+        num_qo_heads = self.num_attention_heads // self.tensor_parallel_shards
+        num_kv_heads = self.num_key_value_heads // self.tensor_parallel_shards
```
Comment (junrushao) [inline python/mlc_chat/model/llama/llama_model.py:362]: minor: we are using `num_q_heads` in MLC so let's be consistent here

**Comment by zazpie** [comment]
Question, If I'm not mistaken this change made `create_tir_paged_kv_cache` needed for everything using the new serve engine.

Since so far only Llama [supports](https://github.com/mlc-ai/mlc-llm/issues/1749) it, is there a way to make it required only for the models using it atm?

**Comment by tqchen** [comment]
we are moving all model defs to use kvcache https://github.com/mlc-ai/mlc-llm/issues/1749

**Pull request merged**